scandium-core/src/test/java/org/eclipse/californium/scandium/ConnectorHelper.java

```diff
@@ -156,16 +159,16 @@ static DtlsConnectorConfig newStandardClientConfig(final InetSocketAddress bindA
 				.setTrustStore(DtlsTestTools.getTrustedCertificates());
 	}
 
-	void givenAnEstablishedSession(final DTLSConnector client) throws Exception {
-		givenAnEstablishedSession(client, true);
+	LatchDecrementingRawDataChannel givenAnEstablishedSession(final DTLSConnector client) throws Exception {
+		return givenAnEstablishedSession(client, true);
 	}
 
-	void givenAnEstablishedSession(final DTLSConnector client, boolean releaseSocket) throws Exception {
+	LatchDecrementingRawDataChannel givenAnEstablishedSession(final DTLSConnector client, boolean releaseSocket) throws Exception {
 		RawData raw = RawData.outbound("Hello World".getBytes(), new AddressEndpointContext(serverEndpoint), null, false);
-		givenAnEstablishedSession(client, raw, releaseSocket);
+		return givenAnEstablishedSession(client, raw, releaseSocket);
 	}
 
-	void givenAnEstablishedSession(final DTLSConnector client, RawData msgToSend, boolean releaseSocket) throws Exception {
+	LatchDecrementingRawDataChannel givenAnEstablishedSession(final DTLSConnector client, RawData msgToSend, boolean releaseSocket) throws Exception {
 
 		CountDownLatch latch = new CountDownLatch(1);
 		LatchDecrementingRawDataChannel clientChannel = new LatchDecrementingRawDataChannel();
```

Comment:
A bit out of topic but what was the purpose of this LatchDecrementingRawDataChannel before you use it in this PR ? :thinking:

Reply:
Forget this, I missed the latch used just after ..